packages/tailwindcss/src/plugin-api.ts

```diff
@@ -346,6 +418,20 @@ function objectToAst(rules: CssInJs | CssInJs[]): AstNode[] {
   return ast
 }
 
+function parseVariantValue(resolved: string | string[], nodes: AstNode[]): AstNode[] {
+  let resolvedArray = typeof resolved === 'string' ? [resolved] : resolved
+  return resolvedArray.flatMap((r) => {
+    if (r.trim().endsWith('}')) {
+      let updatedCSS = r.replace('}', '{@slot}}')
```

Comment:
I don't think this is enough, the moment you have more nesting then only a single `}` would be replaced. I wonder if we can/should use the CSS parser directly on this value 🤔

Reply:
Actually, maybe it is because if you do more nesting like:
```css
.foo {
  .bar {
    .baz {
      &:hover
    } /* <-- this is the one that will be replaced with `{@slot}}` */
  }
} /* <-- this is the main check to determine there is nesting */
```

Then the only invalid one that needs replacing is `&:hover`. The others `.foo`, `.bar` and `.baz` will already be proper CSS rules.

Reply:
The case I was thinking of looks like this: `.foo { .bar { &:hover } .baz { &:hover } }`. Where you have multiple `&:hover` things. But that doesn't even work in  v3

Reply:
Yeah I worked on this with @adamwathan and we decided that this approach, while being quite limited, is enough for V3 parity. I'll add a comment with our explanation but we had something that was more powerful and it felt like total overkill especially since it isn't working in v3 either.